Kinvey.Core/Auth/User.cs

```diff
@@ -583,69 +583,97 @@ static public async Task LoginWithAuthorizationCodeAPIAsync(string username, str
+
+            try
+            {
+                ct.ThrowIfCancellationRequested();
+
+                var clientID = ((KinveyClientRequestInitializer)uc.RequestInitializer).AppKey;
 
                 // Check if a client ID for MIC authentication has been provided
                 //
                 if (!string.IsNullOrEmpty(micID))
-				{
-					clientID += Constants.MIC_ID_SEPARATOR + micID;
-				}
+                {
+                    clientID += Constants.MIC_ID_SEPARATOR + micID;
+                }
 
                 // Initiate token request
                 //
                 var micLoginRequest = BuildMICLogin(uc, username, password, clientID);
-				ct.ThrowIfCancellationRequested();
-				var accessResult = await micLoginRequest.ExecuteAsync();
+                ct.ThrowIfCancellationRequested();
+                var accessResult = await micLoginRequest.ExecuteAsync();
 
-				ct.ThrowIfCancellationRequested();
+                ct.ThrowIfCancellationRequested();
 
-				var accessToken = accessResult["access_token"].ToString();
+                var accessToken = accessResult["access_token"].ToString();
 
-				ct.ThrowIfCancellationRequested();
+                ct.ThrowIfCancellationRequested();
 
-				// Log into Kinvey using the token
-				//
-				var user = await LoginMICWithAccessTokenAsync(accessToken);
+                // Log into Kinvey using the token
+                //
+                var user = await LoginMICWithAccessTokenAsync(accessToken);
 
-				ct.ThrowIfCancellationRequested();
+                ct.ThrowIfCancellationRequested();
 
-				// Store the new access token and refresh token
-				//
-				var currentCred = uc.Store.Load(user.Id, uc.SSOGroupKey);
-				currentCred.AccessToken = accessResult["access_token"].ToString();
-				currentCred.RefreshToken = accessResult["refresh_token"].ToString();
-				currentCred.RedirectUri = uc.MICRedirectURI;
-				currentCred.MICClientID = clientID;
-				uc.Store.Store(user.Id, uc.SSOGroupKey, currentCred);
-			}
-			catch (KinveyException ke)
-			{
-				throw ke;
-			}
-			catch(Exception e)
-			{
-				Logger.Log("Error in LoginWithMIC: " + e.StackTrace);
-				var kinveyException = new KinveyException(EnumErrorCategory.ERROR_USER, EnumErrorCode.ERROR_GENERAL, "Unexpected error in MIC Resource Owner Credentials Grant flow", e);
-				throw kinveyException;
-			}
-		}
+                // Store the new access token and refresh token
+                //
+                var currentCred = uc.Store.Load(user.Id, uc.SSOGroupKey);
+                currentCred.AccessToken = accessResult["access_token"].ToString();
+                currentCred.RefreshToken = accessResult["refresh_token"].ToString();
+                currentCred.RedirectUri = uc.MICRedirectURI;
+                currentCred.MICClientID = clientID;
+                uc.Store.Store(user.Id, uc.SSOGroupKey, currentCred);
+            }
+            catch (KinveyException ke)
+            {
+                throw ke;
```

Comment:
not understanding why we are catching and then throwing the same exception again

Reply:
It was before me and such code exists in different places of our solution.
For example:
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2761_Deprecate_loginWithMIC/Kinvey.Core/Auth/User.cs#L767
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2761_Deprecate_loginWithMIC/Kinvey.Core/Auth/User.cs#L792
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2761_Deprecate_loginWithMIC/Kinvey.Core/Auth/User.cs#L930

I assume we have the behavior to pass the exception to client side.

Reply:
oh I see, so in this case I believe we should do `throw;` instead of `throw ex;`

https://stackoverflow.com/questions/178456/what-is-the-proper-way-to-re-throw-an-exception-in-c

let's create another PR to solve this issue, but you can do this change for the code changes related to this PR